packages/fast-components-styles-msft/src/call-to-action/index.ts

```diff
@@ -86,68 +79,59 @@ export const callToActionButtonOverrides: ComponentStyles<
     },
 };
 
-const styles: ComponentStyles<CallToActionClassNameContract, DesignSystem> = (
-    config: DesignSystem
-): ComponentStyleSheet<CallToActionClassNameContract, DesignSystem> => {
-    const designSystem: DesignSystem = withDesignSystemDefaults(config);
-    const direction: Direction = designSystem.direction;
-    const translateXValue: string = toPx(designSystem.designUnit);
-
-    return {
-        callToAction: {
-            transition: "all 0.2s ease-in-out",
-            display: "inline-flex",
-            maxWidth: "100%",
-            lineHeight: "1",
-            textDecoration: "none",
-            whiteSpace: "nowrap",
-            "&:hover": {
-                "& $callToAction_glyph": {
-                    ...applyGlyphTransform(),
-                },
-            },
-            ...applyFocusVisible("& $callToAction_glyph", {
+const translateXValue: DesignSystemResolver<string> = toPx(designUnit);
+const styles: ComponentStyles<CallToActionClassNameContract, DesignSystem> = {
+    callToAction: {
+        transition: "all 0.2s ease-in-out",
+        display: "inline-flex",
+        maxWidth: "100%",
+        lineHeight: "1",
+        textDecoration: "none",
+        whiteSpace: "nowrap",
+        "&:hover": {
+            "& $callToAction_glyph": {
                 ...applyGlyphTransform(),
-            }),
+            },
         },
-        callToAction_glyph: {
+        ...applyFocusVisible("& $callToAction_glyph", {
+            ...applyGlyphTransform(),
+        }),
+    },
+    callToAction_glyph: {
+        fill: accentForegroundCut,
+        display: "inline-block",
+        position: "relative",
+        width: glyphSize,
+        height: glyphSize,
+        marginLeft: directionSwitch("6px", "unset"),
```

Comment:
Just curious about this new pattern, as we are writing additional style code that's not needed.
`applyLocalizedProperty` would only output the single property, while now with `directionSwitch` we output both `marginRight` and `marginLeft`

Reply:
Yea I realized we can make these empty strings and no value will be written - I'll update.

In general though, yes. The benefit outweighs the cost

Reply:
To elaborate further - this continues on the work to reduce a style-sheets direct dependency on the `DesginSystem` object. There are performance benefits associated with object stylesheets in contrast to function stylesheets (that return stylesheet objects), because we don't need to tear down the `<style>` element when the `designSystem` updates when the stylesheet is an object, like we do when the stylesheet is a function.

Reply:
Awesome, thanks for the explanation.